File: Solutions/70. Climbing Stairs/70.java
class Solution {
    HashMap<Integer,Integer> hm=new HashMap<Integer,Integer>();
    public int climbStairs(int n) {
        if(n==1)return 1;
        if(n==2)return 2;
        if(hm.containsKey(n))return hm.get(n);
        int ans= climbStairs(n-1)+climbStairs(n-2);
        hm.put(n,ans);
        return ans;
    }
}

File: Solutions/70. Climbing Stairs/70.py
class Solution:
    map = {}
    def climbStairs(self, n: int) -> int:
        if n == 1: return 1
        if n == 2: return 2
        if n in self.map:
            return self.map[n]
        ans = self.climbStairs(n-1) + self.climbStairs(n-2)
        self.map[n] = ans
        return ans
        